src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { WorkflowService } from 'app/shared/state/workflow.service';
+import { WorkflowQuery } from 'app/shared/state/workflow.query';
+import { MyEntriesQuery } from 'app/shared/state/my-entries.query';
+import { SessionQuery } from 'app/shared/session/session.query';
+import { SessionService } from 'app/shared/session/session.service';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
+  entryTypeLowerCase: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  totalEntries: number = 0;
+  firstLoad = true;
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    protected userQuery: UserQuery,
+    protected workflowService: WorkflowService,
+    protected workflowQuery: WorkflowQuery,
+    protected myWorkflowsService: MyWorkflowsService,
+    protected myEntriesQuery: MyEntriesQuery,
+    protected sessionQuery: SessionQuery,
+    protected sessionService: SessionService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+    }
+
+    //Get the links for the specified entryType
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
```

Comment:
Too much work for this PR, but I wonder if we shouldn't have sub components to avoid all these if/else conditions, e.g., a workflow-box-component, a tool-box-component, etc.

Reply:
I guess it could be separated into separate components for sure, I'm not against that change later on and it shouldn't be a messy change either.